agent/src/main/java/com/teamscale/jacoco/agent/AgentBase.java

```diff
@@ -45,18 +60,65 @@ public AgentBase(AgentOptions options) throws IllegalStateException {
 
 		logger.info("Starting JaCoCo agent for process {} with options: {}",
 				ManagementFactory.getRuntimeMXBean().getName(), getOptionsObjectToLog());
+		retryUnsuccessfulUploads(options);
 		if (options.getHttpServerPort() != null) {
 			try {
 				initServer();
 			} catch (Exception e) {
-				throw new IllegalStateException(
-						"Control server not started.", e);
+				throw new IllegalStateException("Control server not started.", e);
 			}
 		}
 	}
 
 	/**
-	 * Lazily generated string representation of the command line arguments to print to the log.
+	 * If we have coverage that was leftover because of previously unsuccessful
+	 * coverage uploads, we retry to upload them again with the same configuration
+	 * as in the previous try.
+	 *
+	 * This method expects that the .properties file of the leftover coverage is
+	 * constructed in a specific order.
```

Comment:
This might be a non-obvious behavior that we should document in the README of the agent